@navikt/core/react/src/form/combobox/SelectedOptions/selectedOptionsContext.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import usePrevious from "../../../util/usePrevious";
+import { useInputContext } from "../Input/inputContext";
+import { ComboboxProps } from "../types";
+
+type SelectedOptionsContextType = {
+  selectedOptions: string[];
+  addSelectedOption: (option: string) => void;
+  removeSelectedOption: (option: string) => void;
+  setSelectedOptions: (any) => void;
+  prevSelectedOptions?: string[];
+  singleSelect?: boolean;
+};
+
+const SelectedOptionsContext = createContext<SelectedOptionsContextType>(
+  {} as SelectedOptionsContextType
+);
+
+export const SelectedOptionsProvider = ({
+  children,
+  value,
+}: {
+  children: any;
+  value: Pick<
+    ComboboxProps,
+    "selectedOptions" | "singleSelect" | "onToggleSelected"
+  >;
+}) => {
+  const { setSearchTerm, setValue } = useInputContext();
+  const {
+    selectedOptions: externalSelectedOptions,
+    singleSelect,
+    onToggleSelected,
+  } = value;
+  const [selectedOptions, setSelectedOptions] = useState<string[]>([]);
+
+  useEffect(
+    () => setSelectedOptions(externalSelectedOptions || []),
+    [externalSelectedOptions]
```

Comment:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes